File: AvernusGames_Front-End/src/models/Game.cs
namespace Avernus_Games_Store.src.models
{

    //Construtores gerados pelo chat GPT
    public class Game : Produto
    {
        public Genero? Genero { get; set; }
        public DateTime? ReleaseDate { get; set; }
        public Desenvolvedor? Desenvolvedor { get; set; }
        public Plataforma? Plataforma { get; set; }
        public Game() { } // Construtor vazio
        public Game(string nome, string codProduto, float valorCompra, float markup, string descricao, Genero genero, DateTime releaseDate, Desenvolvedor desenvolvedor, Plataforma plataforma)
            : base(nome, codProduto, valorCompra, markup, descricao)
        {
            Genero = genero;
            ReleaseDate = releaseDate;
            Desenvolvedor = desenvolvedor;
            Plataforma = plataforma;
        } // Construtor com todos os atributos
    }

    public class Genero
    {
        public string? Nome { get; set; }
        public string? Descricao { get; set; }
        public Genero() { } // Construtor vazio
        public Genero(string nome, string descricao)
        {
            Nome = nome;
            Descricao = descricao;
        } // Construtor com todos os atributos
    }

    public class Plataforma
    {
        public string? Nome { get; set; }
        public string? Descricao { get; set; }
        public Plataforma() { } // Construtor vazio
        public Plataforma(string nome, string descricao)
        {
            Nome = nome;
            Descricao = descricao;
        } // Construtor com todos os atributos
    }

    public class Desenvolvedor
    {
        public string? Estudio { get; set; }
        public string? NomeDesenvolvedor { get; set; }
        public Endereco? Endereco { get; set; }
        public Desenvolvedor() { } // Construtor vazio
        public Desenvolvedor(string estudio, string nomeDesenvolvedor, Endereco endereco)
        {
            Estudio = estudio;
            NomeDesenvolvedor = nomeDesenvolvedor;
            Endereco = endereco;
        } // Construtor com todos os atributos
    }
}

File: AvernusGames_Front-End/src/app/models/Game.ts
import { Desenvolvedor } from "./Desenvolvedor";
import { Fornecedor } from "./Fornecedor";
import { Genero } from "./Genero";
import { Plataforma } from "./Plataforma";
export class Game {
    id: number = 0;
    nome: string = 'Desconhecido';
    descricao: string = '';
    valorCompra: number = 0;
    markup: number = 0;
    fornecedor?: Fornecedor
    fornecedorId: number = 0;
    genero?: Genero;
    generoId: number = 0;
    desenvolvedor?: Desenvolvedor;
    desenvolvedorId: number = 0;
    plataforma?: Plataforma;
    plataformaId: number = 0;
    
  }
  